www/js/con-author.js

```diff
@@ -169,16 +259,69 @@ define([
 
 		handleKind: function(kind){
 			console.log("**************** in handleKind ", kind);
+			if(kind == "defaultSelect" || kind == ''){
+				/*
+				this.logging.clientLog("error", {
+					message: "no kind selected for author node type",
+					functionTag: "handleKind"
+				}); */
+				kind = "defaultSelect";
+				this._model.authored.setGenus(this.currentID, kind);
+			}else{
+				this._model.authored.setGenus(this.currentID, kind);
+				this.applyDirectives(this. authorPM.process(this.currentID, "kind", kind));
+			}
+
+			/*
+			this.logging.log('solution-step', {
+				type: "solution-enter",
+				nodeID: this.currentID,
+				property: "kind",
+				node: this._model.given.getName(this.currentID),
+				value: kind
+			});
+			*/
 		},
 
-		handleQuantityDescription: function(description){
+		handleDescription: function(description){
 			// Summary: Checks to see if the given quantity node description exists; if the
 			//		description doesn't exist, it sets the description of the current node.
-		},
+			var descriptionID = this._model.authored.getNodeIDByDescription(description);
+			// If descriptionID is falsy give "null"; if it doesn't match, give "false"
+			var returnObj = this.authorPM.process(descriptionID?!(descriptionID==this.currentID):null, "description", description);
+			console.log("return obj for quantity description", returnObj);
+			this.applyDirectives(returnObj);
+			
+			//var logObj = {};
 
-		handleEquationDescription: function(description){
-			// Summary: Checks to see if the given equation node description exists; if the
-			//		description doesn't exist, it sets the description of the current node.
+			if(!this._model.active.getNodeIDByDescription(description)){
+				this._model.active.setDescription(this.currentID, description);
+				/*
+				logObj = {
+					error: false
+				};
+				*/
```

Comment:
For the above three comments as discussed with @sachingrover211 earlier , a separate function which handles the logging for all the author mode handlers is needed. I think we should make a separate card for this for next week. I discussed this with @jonwetzel briefly